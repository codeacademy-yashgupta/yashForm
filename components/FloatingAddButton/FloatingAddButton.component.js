import React, { Component } from 'react';
import {
  TouchableOpacity, StyleSheet, Image, View,
} from 'react-native';
import icons from '../../public/icons';
import style from './FloatingAddButton.style';

const styles = StyleSheet.create(style);
export default class FloatingAddButton extends Component {
  render() {
    return (
      <View style={styles.addFormIcon}>
        <TouchableOpacity onPress={() => this.props.navigation.navigate('CreateForm')}>
          <Image source={icons.addIcon} style={styles.addImage} />
        </TouchableOpacity>
      </View>
    );
  }
}
