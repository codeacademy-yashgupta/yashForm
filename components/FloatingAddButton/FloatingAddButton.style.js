import { APP_THEME_COLOR } from '../../constants';

export default {
  addImage: { height: 40, width: 40 },
  addFormIcon: {
    position: 'absolute', right: 0, bottom: 0, backgroundColor: APP_THEME_COLOR, borderRadius: 50, marginRight: 30, marginBottom: 30, padding: 15,
  },
};
