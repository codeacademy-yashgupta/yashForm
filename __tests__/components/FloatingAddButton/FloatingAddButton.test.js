import React from 'react';
import renderer from 'react-test-renderer';
import FloatingAddButton from '../../../components/FloatingAddButton/FloatingAddButton.component';
import 'react-native';

describe('FloatingAddButton', () => {
  it('should render correctly', () => {
    const tree = renderer.create(<FloatingAddButton />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
