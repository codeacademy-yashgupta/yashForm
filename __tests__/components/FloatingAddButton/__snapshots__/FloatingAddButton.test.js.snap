// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FloatingAddButton should render correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#3700B3",
      "borderRadius": 50,
      "bottom": 0,
      "marginBottom": 30,
      "marginRight": 30,
      "padding": 15,
      "position": "absolute",
      "right": 0,
    }
  }
>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../public/icons/baseline_add_white_48pt_3x.png",
        }
      }
      style={
        Object {
          "height": 40,
          "width": 40,
        }
      }
    />
  </View>
</View>
`;
